### PR TITLE
Added docstrings on app\examples

### PR DESCRIPTION
**Pull request by ArchaeotheriumSapienter**
Added docstrings on app\examples to tackle #11540 with additional note blocks in some of the files.

### CODE REVIEW

File: examples/app/clustering/main.py [hunk 1]
```diff
@@ -1,3 +1,12 @@
+''' A k-nearest neighbors (KNN) chart using datasets from scikit-learn. This
+example demonstrates solving both classification and regression problems.
+
+.. note::
+    This clustering visualization needs the scikit-learn package to run.
+
+.. _K-nearest neighbors algorithm: https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm
```
Comment (bryevdv) [inline examples/app/clustering/main.py:7]: This link alias does not appear to be used anywhere. The docstring could be updated to include it, or the link anchor could be removed

File: examples/app/weather/main.py [hunk 2]
```diff
@@ -1,3 +1,12 @@
+''' A weather chart for three cities using a csv file.
+This illustration demonstrates different interpretation of the same data
+with the distribution option.
+
+.. note::
+    This chart needs the Scipy and Pandas package to run. See
+    the included `README`_ for more information.
```
Comment (bryevdv) [inline examples/app/weather/main.py:7]:
```suggestion
    ``README.md`` for more information.
```

There's no link alias for README so the trailing underscore has no effect. Suggest using double backticks to render a path code/pre font.

File: examples/app/dash/main.py [hunk 3]
```diff
@@ -1,3 +1,13 @@
+''' A dashboard that utilizes `Auto MPG`_ and stocks dataset.
+This example demonstrates the use of Bootstrap templates to compile several types
+of charts in one file.
+
+.. note::
+    This chart needs the Pandas package to run.
+
+.. _Auto MPG dataset: https://archive.ics.uci.edu/ml/datasets/auto+mpg
```
Comment (bryevdv) [inline examples/app/dash/main.py:8]: This link alias here does not match what is used above in single backticks in the docstring. They have to match exactly or Sphinx will not generate a link in the rendered docs

**Review by bryevdv — CHANGES_REQUESTED** [review]

**Comment by bryevdv** [comment]
Thanks @ArchaeotheriumSapienter just a few final suggestions concerning Sphinx link alias (please push new changes to this same branch)

**Comment by ArchaeotheriumSapienter** [comment]
I'm done with the changes in this Pull Request, thank you so much!

File: examples/app/faces/main.py [hunk 4]
```diff
@@ -1,3 +1,13 @@
+''' A face detection example that uses the Haar Cascade algorithm.
+This example shows the capability of Bokeh streaming with an integrated
+OpenCV package.
+
+.. note::
+    This chart needs the OpenCV package to run.
+
+.. _Haar Cascade: https://docs.opencv.org/3.4/db/d28/tutorial_cascade_classifier.html
```
Comment (bryevdv) [inline examples/app/faces/main.py:8]: This alias is not used in the text

File: examples/app/image_blur.py [hunk 5]
```diff
@@ -1,3 +1,10 @@
+''' An image blurring example. This sample shows the capability
+of Bokeh to transform images to have certain effects.
+
+.. note::
+    This file needs the scipy package to run.
```
Comment (bryevdv) [inline examples/app/image_blur.py:5]:
```suggestion
    This example needs the scipy package to run.
```

File: examples/app/simple_hdf5/main.py [hunk 6]
```diff
@@ -1,3 +1,9 @@
+''' A simple chart visualizing hdf5 files.
+
+.. note::
+    This visualization needs the hdf5 library to run.
```
Comment (bryevdv) [inline examples/app/simple_hdf5/main.py:4]:
```suggestion
    This example needs the hdf5 library to run.
```
(let's use "example" verbiage consistently)

File: examples/app/spectrogram/main.py [hunk 7]
```diff
@@ -1,3 +1,10 @@
+''' A spectogram chart that uses a waterfall dataset.
+This example shows the streaming efficiency of Bokeh with live audio.
+
+.. note::
+    This clustering visualization needs the scipy package to run.
```
Comment (bryevdv) [inline examples/app/spectrogram/main.py:5]:
```suggestion
    This example needs the scipy package to run.
```

File: examples/app/clustering/main.py [hunk 8]
```diff
@@ -1,3 +1,12 @@
+''' A `k-nearest neighbors`_ (KNN) chart using datasets from scikit-learn. This
+example demonstrates solving both classification and regression problems.
+
+.. note::
+    This clustering visualization needs the scikit-learn package to run.
```
Comment (bryevdv) [inline examples/app/clustering/main.py:5]:
```suggestion
    This example needs the scikit-learn package to run.
```

File: examples/app/crossfilter/main.py [hunk 9]
```diff
@@ -1,3 +1,13 @@
+''' A crossfilter plot map that uses the `Auto MPG dataset`_. This example
+demonstrates the relationship of datasets together. A hover tooltip displays
+information on each dot.
+
+.. note::
+    This chart needs the Pandas package to run.
```
Comment (bryevdv) [inline examples/app/crossfilter/main.py:6]:
```suggestion
    This example needs the Pandas package to run.
```

File: examples/app/dash/main.py [hunk 10]
```diff
@@ -1,3 +1,13 @@
+''' A dashboard that utilizes `Auto MPG`_ and stocks dataset.
+This example demonstrates the use of Bootstrap templates to compile several types
+of charts in one file.
+
+.. note::
+    This chart needs the Pandas package to run.
```
Comment (bryevdv) [inline examples/app/dash/main.py:6]:
```suggestion
    This example needs the Pandas package to run.
```

File: examples/app/faces/main.py [hunk 11]
```diff
@@ -1,3 +1,13 @@
+''' A face detection example that uses the Haar Cascade algorithm.
+This example shows the capability of Bokeh streaming with an integrated
+OpenCV package.
+
+.. note::
+    This chart needs the OpenCV package to run.
```
Comment (bryevdv) [inline examples/app/faces/main.py:6]:
```suggestion
    This example needs the OpenCV package to run.
```

**Review by bryevdv — COMMENTED** [review]
Last comments: 

* one unused alias 
* edits to use consistent verbiage (always good to be consistent this way, to help situate users)

**Comment by bryevdv** [comment]
FYI if you go the files tab, you can apply suggestions in a batch from the web UI (if you want to use that method)

**Comment by ArchaeotheriumSapienter** [comment]
Thank you for the input! I'll keep those in mind next time :)

File: examples/app/weather/main.py [hunk 12]
```diff
@@ -1,3 +1,12 @@
+''' A weather chart for three cities using a csv file.
+This illustration demonstrates different interpretation of the same data
+with the distribution option.
+
+.. note::
+    This chart needs the Scipy and Pandas package to run. See
```
Comment (bryevdv) [inline examples/app/weather/main.py:6]: Sorry I guess I missed one last one, will be good to merge after this:

```suggestion
    This example needs the Scipy and Pandas package to run. See
```

Comment (ArchaeotheriumSapienter) [reply examples/app/weather/main.py:6]: No worries! Its also on my part since I was not consistent, I'll keep your suggestions in mind next time :)

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @ArchaeotheriumSapienter !

**Comment by ArchaeotheriumSapienter** [comment]
Thank you!

**Pull request merged**